[edoc] remove functionality related to packages

vladdu opened the pull request:
Packages have been removed trom the language. There's no need to have code supporting them.

The 'subpackages' option is left, since some projects use recursive source directories.

marcusarendt commented:
Problems detected in our nightly builds:
edoc:toc/3 calls undefined edoc_lib:get_doc_env/4
edoc_extract:collect/7 calls undefined erl_syntax:rule_clauses/1

vladdu commented:
Rebased on latest master and fixed the edoc_lib:get_doc_env problem.

The other one is not related to this issue, that code wasn't touched and erl_syntax:rule_classes exists.

OTP-Maintainer commented:
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------
